eth-contracts/contracts/service/ServiceProviderFactory.sol

```diff
@@ -475,12 +529,15 @@ contract ServiceProviderFactory is RegistryContract {
     function isValidServiceType(bytes32 _serviceType)
     external view returns (bool isValid)
     {
+        return serviceTypeStakeRequirements[_serviceType].maxStake > 0;
```

Comment:
going with previous comment of changing this to serviceTypeinfo, this check could then become serviceTypeInfo[_serviceType].index > 0 (you'd have to 1-index the array)